File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder sb = new StringBuilder();
        
        for (char c : s.toCharArray()) {
            if (Character.isAlphabetic(c) || Character.isDigit(c)) {
                sb.append(c);
            }
        }
        
        String sbString = sb.toString().toLowerCase();
        
        
        return sbString.equals(sb.reverse().toString().toLowerCase());
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        l = 0
        r = len(s) - 1
        
        while l < r:
            while l < r and not self.isCharOrDig(s[l]):
                l += 1
            while r > l and not self.isCharOrDig(s[r]):
                r -= 1
            if s[l].lower() != s[r].lower():
                return False
            
            l = l + 1
            r = r - 1
        
        return True
    
    
    def isCharOrDig(self, c: chr) -> bool:
        return (ord('A') <= ord(c) <= ord('Z') or
        ord('a') <= ord(c) <= ord('z') or
        ord('0') <= ord(c) <= ord('9'))